src/main/java/racingcar/service/RacingService.java

```diff
@@ -0,0 +1,63 @@
+package racingcar.service;
+
+import racingcar.domain.Car;
+import racingcar.domain.Cars;
+import racingcar.domain.dto.OutputDto;
+import racingcar.domain.movable.RandomForwardStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RacingService {
+
+    private final Cars cars;
+    private List<OutputDto> dtos = new ArrayList<>();
+
+    private RacingService(final Cars cars) {
+        this.cars = cars;
+    }
+
+    public static RacingService from(final Cars cars) {
+        return new RacingService(cars);
+    }
+
+    public void drive() {
+        IntStream.range(0, cars.getCars().size())
+            .forEach(idx -> {
+                cars.getCars().get(idx).moveForward();
+                OutputDto dto = new OutputDto(cars.getCars().get(idx));
+                dto.convertCurrentCarStatement();
+                dtos.add(dto);
+            });
+    }
+
+    public List<Car> notifyCurrentStatement() {
+        return cars.getCars();
+    }
+
+    public List<String> judgeWinners() {
+        return cars.getCars().stream()
+            .filter(car -> (car.getStep() == maxDriveLength() && car.getStep() > 0))
```

Comment:
지금 코드는 루프를 돌때마다 `maxDriveLength()`를 호출하게 됩니다.

```suggestion
        maxPositition = maxDriveLength()
        return cars.getCars().stream()
            .filter(car -> (car.getStep() == maxPositition && car.getStep() > 0))
```

Reply:
확인했습니다 stream안에서 계속 반복 호출되고 있었군요!